src/extension.ts

```diff
@@ -1731,9 +1736,34 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
     }
 }
 
-async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+async function useNamespaceKubernetes(explorerNode?: explorer.KubernetesObject) {
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
+            refreshExplorer();
+        }
+    } else {
+        promptKindName([kuberesources.allKinds.namespace], 'get', undefined, (resource) => {
+            if (resource) {
+                const currentNS = kubectlUtils.currentNamespace(kubectl);
+                currentNS.then((ns) => {
+                    //resource will be of format <kind>/<name>, hence get the substring
+                    let toSwitchNamespace = '' + resource;
```

Comment:
Is this to coerce `resource` to a string?  Surely `promptKindName` produces a `string | undefined` so no coercion is needed?  (We should add type annotations to the `promptKindName.handler` parameter to have it checked though.)